### PR TITLE
Whitelist slackbot by id - poke

### PR DESCRIPTION
**Pull request by albandum**
## Description

Add the possibility to whitelist a slack worflow using its ID through poke

https://github.com/user-attachments/assets/0a5bcd33-1040-4c84-9b82-c093ac9e35cd

## Risk

Not much - it's poke only. The real risk is on https://github.com/dust-tt/dust/pull/7735

## Deploy Plan

Wait for https://github.com/dust-tt/dust/pull/7735 to be deployed

### CODE REVIEW

File: front/pages/poke/[wId]/data_sources/[dsId]/index.tsx [hunk 1]
```diff
@@ -596,7 +598,10 @@ async function handleWhitelistBot({
       majorCommand: "slack",
       command: "whitelist-bot",
       args: {
-        botName,
+        botNameOrId,
+        ...(selectedBotIdentifierType === "name"
+          ? { botName: botNameOrId }
```
Comment (philipperolet) [inline front/pages/poke/[wId]/data_sources/[dsId]/index.tsx:603]: :100: :wink:

File: front/pages/poke/[wId]/data_sources/[dsId]/index.tsx [hunk 2]
```diff
@@ -596,7 +598,10 @@ async function handleWhitelistBot({
       majorCommand: "slack",
       command: "whitelist-bot",
       args: {
-        botName,
+        botNameOrId,
```
Comment (philipperolet) [inline front/pages/poke/[wId]/data_sources/[dsId]/index.tsx:601]: this arg can be removed, it's not used by the endpoint (only botName below is)

File: front/pages/poke/[wId]/data_sources/[dsId]/index.tsx [hunk 3]
```diff
@@ -849,12 +857,31 @@ function SlackWhitelistBot({
   return (
     <div className="mb-2 flex flex-col gap-2 rounded-md border px-2 py-2 text-sm text-gray-600">
       <div className="flex items-center gap-2">
-        <div>Whitelist slack bot or workflow</div>
+        <div>Whitelist slack bot or workflow by</div>
+        <DropdownMenu>
+          <DropdownMenu.Button
+            label={selectedBotIdentifierType ?? "Whitelist type"}
```
Comment (philipperolet) [inline front/pages/poke/[wId]/data_sources/[dsId]/index.tsx:863]: typing above prevents this state var from being null/undefined no?

**Review by philipperolet — APPROVED** [review]
Thanks ! Looks good :) Left 2 nits but pre-approving

We can indeed wait for other PR before merging; may I suggest to be extra clear on bot not found in said other PR? because I suspect users (us) won't pay attention and forget to switch to the right type, and then wonder why the bot is not found :sweat_smile:

**Comment by albandum** [comment]
> I suspect users (us) won't pay attention and forget to switch to the right type, and then wonder why the bot is not found 😅

Modified a bit to make sure the user has to chose (no default set, warning displayed if no choice made)

**Pull request merged**